syntax = "proto2";

message RoomRequest {
    message Join {
        repeated bytes  miis      = 1;
        required uint32 location  = 2;
        required uint32 latitude  = 3;
        required uint32 longitude = 4;
    }

    message Spectate {}

    oneof request {
        Join     join     = 1;
        Spectate spectate = 2;
    }
}

message RoomEvent {
    message Join {
        required bytes  mii       = 1;
        required uint32 location  = 2;
        required uint32 latitude  = 3;
        required uint32 longitude = 4;
    }

    message Leave {
        required uint32 playerId = 1;
    }

    message Spectate {
        required uint32 count = 1;
    }

    message Host {
        required uint32 playerId = 1;
    }

    oneof event {
        Join     join     = 1;
        Leave    leave    = 2;
        Spectate spectate = 3;
        Host     host     = 4;
    }
}
